core/domain/summary_services.py

```diff
@@ -222,3 +223,32 @@ def get_featured_exploration_summary_dicts(language_codes):
         reverse=True)
 
     return _get_displayable_exp_summary_dicts(sorted_exp_summaries)
+
+
+def get_top_rated_exploration_summary_dicts(language_codes):
+    """Returns a list of top rated explorations with the given language code.
+
+    The return value is sorted in decreasing order of search rank.
+    """
+    filtered_exp_summaries = [
+        exp_summary for exp_summary in
+        exp_services.get_non_private_exploration_summaries().values()
+        if exp_summary.language_code in language_codes]
+
+    average_rankings = {
+        exp_summary.id: exp_services.get_average_rank_from_exp_summary(
+            exp_summary)
+        for exp_summary in filtered_exp_summaries
+    }
+
+    sorted_by_ranking_count_exp_summaries = sorted(
```

Comment:
Nice use of stable sort. Suggest adding a comment above this: "These two calls to 'sorted' ensure that the return list is sorted by average rating, breaking ties by the number of ratings."